test/filters/test_convolution.cpp

```diff
@@ -3155,6 +3171,151 @@ main (int argc, char** argv)
+
+  std::uint32_t j = 0;
+  for (std::uint32_t i = 0; i < output->width ; ++i)
+    EXPECT_NEAR ((*output) (i,j).intensity, output_0[i], 1e-3);
+  j = 47;
+  for (std::uint32_t i = 0; i < output->width ; ++i)
+    EXPECT_NEAR ((*output) (i,j).intensity, output_47[i], 1e-3);
+}
+
+/// \todo Correct input values and filter factors required to test the function and not just detect compiler errors.
+TEST (Convolution, convolveRowsRGB)
+{
+  using namespace pcl::common;
+  using namespace pcl::filters;
+
+  PointCloud<RGB>::Ptr input (new PointCloud<RGB> ());
+  Eigen::ArrayXf filter(7);
+  pcl::filters::Convolution<RGB, RGB> convolve;
+  PointCloud<RGB>::Ptr output;
+
+  // input
+  filter << 0.00443305, 0.0540056, 0.242036, 0.39905, 0.242036, 0.0540056, 0.00443305;
+  input->width = 48;
+  input->height = 48;
+  input->resize(input->width * input->height);
+  for (std::uint32_t r = 0; r < input->height; r++)
+    for (std::uint32_t c = 0; c < input->width; c++)
+    {
+      float x1 = -2.0f + (4.0f / (float)input->width) * (float)c;
+      float y1 = -2.0f + (4.0f / (float)input->height) * (float)r;
+      float x2 = -M_PI + (2.0f * M_PI / (float)input->width) * (float)c;
+      float y2 = -2.0f + (4.0f / (float)input->height) * (float)r;
+      float z = x1 * exp(-(x1 * x1 + y1 * y1)) * 2.5f + sin(x2) * sin(y2);
+      COLOR color = get_color((COLOR*)basemap, 5, -1.6f, 1.6f, z);
+      (*input) (r, c).r = (uint8_t)(255.0f * color.r);
+      (*input) (r, c).g = (uint8_t)(255.0f * color.g);
+      (*input) (r, c).b = (uint8_t)(255.0f * color.b);
+    }
+
+  // filter
+  output.reset (new PointCloud<RGB> ());
+  convolve.setInputCloud (input);
+  convolve.setKernel (filter);
+  convolve.convolveRows (*output);
+
+  // first output row
+  std::uint32_t i = 0;
+  for (std::uint32_t j = 0; j < output->width; ++j)
+  {
+    EXPECT_NEAR((*output) (i, j).r, 0, 1e-3);
+    EXPECT_NEAR((*output) (i, j).g, 0, 1e-3);
+    EXPECT_NEAR((*output) (i, j).b, 0, 1e-3);
```

Comment:
Same as above

Reply:
In my tests I can use EXPECT_FLOAT_EQ I have changed that.